runtime/src/iree-amd-aie/driver/xrt/cts/CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 The IREE Authors
```

Comment:
Why we need to add a CTS test for XRT HAL driver in this PR?

This not mentioned in PR description. What does the CTS testdo?

Refer to the PR description below:

```
Let's try this again: this is a HAL that skips/avoids/elides XRT completely. It does so by (essentially) going straight to the [XDNA driver's (kmq) shim](https://github.com/amd/xdna-driver/tree/main/src/shim) which talks directly to the kernel module. Note that XDNA's shim actually couples tightly to XRT itself so the bulk of the actual "work" here is/was decoupling. Thus, we vendor the decoupled shim. Just to be explicit/clear: after this PR, on Linux, the only thing needed in the environment is the [kernel module itself](https://github.com/amd/xdna-driver/tree/main/src/driver/amdxdna) - not XRT, nor XDNA's shim.

Call outs:

Only Linux for now (Windows soon);
On Linux, we no longer even need to clone XRT (completely really does mean completely);
We no longer use .xclbin as the artifact and now just load the .pdi directly;
This uses the so-called kernel-managed queue (kmq) path; user-managed queues (umq) aren't supported until aie4;
The current implementation here is still sync and still uses direct (rather than indirect) command buffers; this PR is already large/complex enough without changing the paradigms. The next PR will add/implement those;
Nothing about how drivers themselves are bumped needs to change - you/anyone can continue to install the XDNA drivers using their full instructions there and everything will work the same, i.e., the XRT pieces will still be skipped even if you have them installed. But I will probably put together a script that builds/installs only their kernel module for those that want to opt-out of XRT in their environment;
For now, the XRT HAL will still work on Linux but it should be considered deprecated (hence no longer tested in CI).
TODOs before landing:

 remove instructional comments leftover from the null HAL;
 undo "OO"-ness of the HAL
 add missing trace zones
 remove smart pointers where possible
 move hwq into kernel
 remove exceptions from the shim.
 parameterize number of cols/rows (currently hardcoded 4x4 for Phoenix);
```